test/puppetlabs/puppetdb/testutils/jetty.clj

```diff
@@ -60,16 +60,16 @@
     (spit logback-file (log-config log-file "ERROR"))
     [log-file (assoc-in config [:global :logging-config] logback-file)]))
 
-(defn current-port
-  "Given a trapperkeeper server, return the port of the running jetty instance.
-  Note there can be more than one port (i.e. SSL + non-SSL connector). This only
-  returns the first one."
-  [server]
-  (-> @(tka/app-context server)
-      (get-in [:WebserverService :jetty9-servers :default :server])
-      .getConnectors
-      first
-      .getLocalPort))
+(defn open-port-num
+  "Returns a currently open port number"
+  []
+  (with-open [s (java.net.ServerSocket. 0)]
```

Comment:
I talked with @mullr about this. The issue is the source of a PuppetDB replication needs to know it's own port number so it can send it's URL to the remote (destination) of the replication. If the port isn't determined until startup, you can't go back and change the config (config has already started). Switching to this model lets the startup happen normally with the port already set.

Also there seems to be some state kept by the server socket, i.e. opening one port, then closing and opening it again picks a different port.